refactoringminer/src/main/java/gr/uom/java/xmi/decomposition/PsiUtils.java

```diff
@@ -29,16 +29,9 @@ public static boolean isTypeKeyword(PsiElement keyword) {
         return PsiUtil.isJavaToken(keyword, typeKeywords);
     }
 
-    public static PsiJavaToken findLastCArrayBracket(PsiElement element) {
-        PsiJavaToken last = null;
-        PsiElement current = element;
-        while (current != null) {
-            if (PsiUtil.isJavaToken(current, JavaTokenType.RBRACKET)) {
-                last = (PsiJavaToken) current;
-            }
-            current = current.getNextSibling();
-        }
-        return last;
+    @SuppressWarnings("unchecked")
+    public static <T> T findFirstChildOfType(PsiElement element, Class<T> type) {
```

Comment:
why `Class<T> type`, not `PsiElement type`?

Reply:
Because we need Java type of searched element for checking by `instanceof`